www/src/components/auth/AuthenticateButton.tsx

```diff
@@ -0,0 +1,39 @@
+import { useAuth0 } from '@auth0/auth0-react';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+type BaseButtonProps = {
+    text: string;
+    onClick: () => void;
+};
+
+const BaseButton: React.FC<BaseButtonProps> = (props) => {
+    return <button onClick={props.onClick}>{props.text}</button>;
+};
+
+const LogoutButton = () => {
+    const { logout } = useAuth0();
+    return <BaseButton text="Logout" onClick={() => logout()} />;
+};
+
+const LoginButton = () => {
+    const { loginWithRedirect } = useAuth0();
+    return <BaseButton text="Login" onClick={() => loginWithRedirect()} />;
+};
+
+const AuthenticateButton: React.FC = () => {
+    const { isAuthenticated } = useAuth0();
+
+    if (isAuthenticated) {
+        return <LogoutButton />;
+    }
+
+    return <LoginButton />;
+};
+
+BaseButton.propTypes = {
+    text: PropTypes.string.isRequired,
+    onClick: PropTypes.func.isRequired,
+};
```

Comment:
Asking b/c I'm naive with TS and React. Are prop types still needed when you have the typing of TypeScript already?

Reply:
This is a `react-eslint` issue. There's an online debate whether this is needed. I personally would like to disable that rule if possible :rofl:

Reply:
Okay well FWIW if we have the props explicitly typed somewhere with TS then I'm fine to remove it.